activerecord/lib/arel/visitors/postgresql.rb

```diff
@@ -36,6 +36,45 @@ def visit_Arel_Nodes_NotRegexp(o, collector)
           infix_value o, collector, op
         end
 
+        def visit_Arel_Nodes_HomogeneousIn(o, collector)
+          if o.type == :in && o.attribute.type_caster.is_a?(ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Point)
```

Comment:
Arel should not know about the classes inside the `ActiveRecord` namespace. Is checking the type caster the only way we have to know the attribute is a point? Is that is the case maybe we need to generate a new node in the Active Record relation, or the condition builder.

Reply:
>  Is checking the type caster the only way we have to know the attribute is a point?

`ActiveRecord::Point` is just a struct (defined [here](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/postgresql/oid/point.rb#L4)). So I suppose you could check if it responds to `:x` and `:y`. That doesn't feel right though.

Is your advice to add a new node (https://github.com/rails/rails/tree/main/activerecord/lib/arel/nodes) instead?

> Arel should not know about the classes inside the ActiveRecord namespace.

Should Arel only know about things in the `Arel` namespaces? It references ActiveModel [here](https://github.com/rails/rails/blob/e32eb318c2daf8de05161c82ddade9e4f4adbddf/activerecord/lib/arel/visitors/to_sql.rb#L625); I can try and tidy that up if needed.

Reply:
@rafaelfranca I have updated the code, I'm not sure it is exactly what you were suggesting though. If it is not, can you provide me any guidance about the correct/expected approach?

FYI @rafaelfranca and @ghiculescu, there are a couple of places that reference `ActiveModel` within `arel`, I have tracked that, and seems that dependency was added on purpose, the commit message says that as now arel is part of the rails code, we can do it